### ISSUE TITLE
Preventing removal of existing dependencies by installing them again

### ISSUE THREAD

**Issue by braham-snyder** [jorgebucaran/fisher#388]
Currently, it appears that removing a plugin removes each of its dependencies as well, for each dependency not already installed prior to the installation of the dependent plugin about to be removed. Perhaps, ideally, `install`ing a plugin already installed as a dependency of another plugin could also prevent that dependency from being removed upon removal of the dependent plugin.

This is the current behavior I'm seeing for the latter case, where attempting to install an existing dependency plugin does not prevent its later, indirect removal when uninstalling the dependent package:
```
⋊> fisher install fzf
Installing 1 plugin/s
OK Copy fzf
Installing 2 dependencies
OK Copy await
OK Copy get
OK Copy last_job_id
OK Copy getopts
⋊> fisher install last_job_id
No plugins to install or dependencies missing.
⋊> fisher rm fzf
fish: Unknown command 'last_job_id'
~/.config//fish/functions/fish_right_prompt.fish (line 1):
last_job_id -l
^
in command substitution
        called on line 0 of file ~/.config//fish/functions/fish_right_prompt.fish

in function 'fish_right_prompt'
        called on standard input

in command substitution
        called on standard input
```

**Comment by jbucaran** [comment]
@braham-snyder Hi. I am not sure I fully get what you are saying (sorry! 😅), but by looking at the console output I think what's happening is that your prompt (specifically your right prompt) depends on last_job_id and if you remove this plugin, the prompt breaks. 

Notice that, if you remove a plugin that depends on last_job_id, last_job_id won't get uninstalled along if there's another plugin that still depends on last_job_id. We track that on `$fisher_dependency_count`.

**Comment by jbucaran** [comment]
> @braham-snyder Perhaps, ideally, installing a plugin already installed as a dependency of another plugin could also prevent that dependency from being removed upon removal of the dependent plugin.

I think I see what you are saying. Instead of complicating the UI, I'd rather make sure `$fisher_dependency_count` works the way it is supposed to work assuming it is buggy.

**Comment by braham-snyder** [comment]
My apologies, my example almost certainly would have been better had I attempted to just call `last_job_id` manually.

> I think I see what you are saying. Instead of complicating the UI, I'd rather make sure $fisher_dependency_count works the way it is supposed to work assuming it is buggy.

I don't think I'd call `$fisher_dependency_count` buggy, if you're asking that -- to clarify: I'm just thinking it might be useful (though more complicated, as you mention) to allow the "promotion" of installed dependencies ("promotion" as used [here](https://github.com/Homebrew/brew/pull/1813)).

I completely understand if you'd like to close this as beyond the scope of `fisher` -- just thought it was worth raising.

**Comment by jbucaran** [comment]
@braham-snyder No worries. $fisher_dependency_count must be buggy, otherwise you should not be getting that error after you uninstall fzf.

Installing a prompt that has `last_job_id` adds +1 to `last_job_id` in `$fisher_dependency_count`. +1 from installing fzf and you get 2. Uninstalling fzf, should not have caused fisherman to uninstall `last_job_id` prematurely.

**Comment by braham-snyder** [comment]
Sorry about that, I believe that was just another flaw in my (quite poor, in retrospect) example: I manually removed `last_job_id` before that excerpt while toying with `fisher`'s dependency management. I really should have chosen a smaller example.

**Comment by jbucaran** [comment]
@braham-snyder It's cool. Do you have another example? 

I understand the feature request (and it also seems relatively simple to implement), but one of the reasons `$fisher_dependency_count` was introduced is so that users don't have to manually update a plugin's internal ref count.

**Comment by jorgebucaran** [comment]
@braham-snyder Fixed in [fisher/releases/tag/3.0.0](https://github.com/jorgebucaran/fisher/releases/tag/3.0.0)~! 🎉 

---

For more information see the [new documentation](https://github.com/jorgebucaran/fisher/blob/34b5257c5beaaf37f7d1988a6a9a62f8a0b1196c/README.md).
